src/vs/workbench/contrib/notebook/browser/contrib/find/findFilters.ts

```diff
@@ -70,29 +72,55 @@ export class NotebookFindFilters extends Disposable {
 		}
 	}
 
-	private _searchInRanges: boolean = false;
+	private _findInSelection: boolean = false;
 
-	get searchInRanges(): boolean {
-		return this._searchInRanges;
+	get findInSelection(): boolean {
+		return this._findInSelection;
 	}
 
-	set searchInRanges(value: boolean) {
-		if (this._searchInRanges !== value) {
-			this._searchInRanges = value;
-			this._onDidChange.fire({ searchInRanges: value });
+	set findInSelection(value: boolean) {
+		if (this._findInSelection !== value) {
+			this._findInSelection = value;
+			this._onDidChange.fire({ findInSelection: value });
 		}
 	}
 
-	private _selectedRanges: ICellRange[] = [];
+	private _findScopeType: NotebookFindScopeType = NotebookFindScopeType.Cells;
 
-	get selectedRanges(): ICellRange[] {
-		return this._selectedRanges;
+	get findScopeType(): NotebookFindScopeType {
+		return this._findScopeType;
 	}
 
-	set selectedRanges(value: ICellRange[]) {
-		if (this._selectedRanges !== value) {
-			this._selectedRanges = value;
-			this._onDidChange.fire({ searchInRanges: this._searchInRanges });
+	set findScopeType(value: NotebookFindScopeType) {
+		if (this._findScopeType !== value) {
+			this._findScopeType = value;
+			this._onDidChange.fire({ findInSelection: this._findInSelection });
+		}
+	}
+
+	private _selectedCellRanges: ICellRange[] | undefined = undefined;
+
+	get selectedCellRanges(): ICellRange[] | undefined {
+		return this._selectedCellRanges;
+	}
+
+	set selectedCellRanges(value: ICellRange[] | undefined) {
+		if (this._selectedCellRanges !== value) {
```

Comment:
should we do range comparison here other than value check? they are not primitive types so it will always be false.